### PR TITLE
Add --include-hidden flag

### PR DESCRIPTION
**Pull request by xtyrrell**
Adds the `--include-hidden` flag.

- [x] Did you add adequate test and make sure they pass by running `npm run test` Yes
- [x] Did you add update docs in the `README.md` file? Yes

Changes proposed in this pull request:
- Adds the `--include-hidden` flag. This allows the user to tell the compiler to not ignore files beginning with a '.' or a '_'.

### CODE REVIEW

**Comment by ImFeelingDucky** [comment]
My intention with this feature was to allow the `less-watch-compiler` just to watch hidden files, and cause non-hidden files to recompie on changes to the hidden files.

For example, in a two-file repository:
```
main.less
_variables.less
```
where `main.less` imports `_variables.less`, I want to be able to edit `_variables.less` and have those changes cause `main.less` to recompile and output a new `main.css` file.

This PR is flawed and does not do that: it actually outputs the `_variables.css` file.

**Comment by jonycheung** [comment]
@xtyrrell This PR is awesome! It has tests and everything. I should've looked at this sooner. Merging and releasing it now.

**Comment by jonycheung** [comment]
Just saw your comment. I'll have a look and see if I can resolve that side effect.

File: src/less-watch-compiler.js [hunk 1]
```diff
@@ -116,7 +118,7 @@ function init(){
     console.log('Watching directory for file changes.');
   lessWatchCompilerUtils.watchTree(
     lessWatchCompilerUtils.config.watchFolder,
-    {interval: 200, ignoreDotFiles: true, filter:lessWatchCompilerUtils.filterFiles},
+    {interval: 200, ignoreDotFiles: true, filter:lessWatchCompilerUtils.filterFiles(lessWatchCompilerUtils.config.includeHidden)},
```
Comment (jonycheung) [inline src/less-watch-compiler.js]: We don't need to pass the config here I think. It's causing some error

**Review by jonycheung — COMMENTED** [review]
Thanks! So good.

**Comment by jonycheung** [comment]
Merging this in so I can raise another PR

**Pull request merged**